src/storage-client/src/controller/collection_mgmt.rs

```diff
@@ -78,27 +88,101 @@ impl CollectionManager {
                             }
                         }
                     },
-                    cmd = rx.recv() => {
-                        if let Some((id, updates)) = cmd {
-                            assert!(collections.lock().await.contains(&id));
+                    cmd = rx.recv_many(64) => {
+                        if let Some(batch) = cmd {
+                            #[derive(Default)]
+                            struct UpdateRequest {
+                                rows: Vec<(Row, Diff)>,
+                                notifs: Vec<oneshot::Sender<Result<(), StorageError>>>,
+                            }
+
+                            // Group all of our updates based on ID.
+                            let mut updates: BTreeMap<GlobalId, UpdateRequest> = BTreeMap::new();
+                            for (id, rows, notif) in batch {
+                                let request = updates.entry(id).or_default();
+                                request.rows.extend(rows);
+                                request.notifs.push(notif);
+                            }
+
+                            // Make sure all of the collections exist.
+                            let collections = collections.lock().await;
+
+                            let (mut updates, non_existent): (BTreeMap<_, _>, BTreeMap<_, _>) = updates
+                                .into_iter()
+                                .partition(|(key, _val)| collections.contains(key));
+                            drop(collections);
+
+                            // Return errors for requests whose collection does not exist.
+                            for (id, UpdateRequest { notifs, .. }) in non_existent {
+                                for notif in notifs {
+                                    // We don't care if the listener disappeared.
+                                    let _ = notif.send(Err(StorageError::IdentifierMissing(id)));
```

Comment:
I think this should likely be its own error message; the case is not necessarily that the identifier is missing (though that's possible), but it might also have never been registered as a managed collection. Can't realistically know which so the error should reflect that in some way.